pkg/controller/conformance/runtime.go

```diff
@@ -564,3 +564,149 @@ func (suite *RuntimeSuite) TestControllerRuntimeMetrics() {
 		assert.Equal(collect, getIntValFromExpVarMap(metrics.ControllerCrashes, ctrl.Name()), 1, "ControllerCrashes should be 1")
 	}, 10*time.Second, 10*time.Millisecond)
 }
+
+// TestQIntToStrController ...
+func (suite *RuntimeSuite) TestQIntToStrController() {
+	srcNS := "q-int"
+	targetNS := "q-str"
+
+	suite.Require().NoError(suite.Runtime.RegisterQController(&QIntToStrController{
+		SourceNamespace: srcNS,
+		TargetNamespace: targetNS,
+	}))
+
+	suite.Require().NoError(suite.State.Create(suite.ctx, NewIntResource(srcNS, "id1", 1)))
+	suite.Require().NoError(suite.State.Create(suite.ctx, NewIntResource(srcNS, "id2", 2)))
+
+	suite.startRuntime()
+
+	suite.Assert().NoError(retry.Constant(10*time.Second, retry.WithUnits(10*time.Millisecond)).Retry(
```

Comment:
can we use `rtestutils.AssertResources` here?

Reply:
no, these resources don't support ResourceWithRD

Reply:
ok I see